qtcodes/circuits/circ.py

```diff
@@ -40,36 +39,41 @@ def __init__(
                 Prepended to all registers.
 
         """
-        self.params = params if params else {"d": 3}
+        self.params = params if params else [{"d": 3}]*len(ctype)
         self.name = name
-
         self.n = 0
-
         # == None is necessary, as "not circ" is true for circ=QuantumCircuit()
         self.circ = QuantumCircuit() if circ is None else circ
-
-        if ctype not in blueprint:
-            raise ValueError(
-                "Please choose a Topological Qubit type from: "
-                + str(list(blueprint.keys()))
-            )
-        self.tqubit_type = blueprint[ctype]
+        self.tqubit_types = []
         self.tqubits: Dict[str, Dict[int, Type[Any]]] = {}
-        for _ in range(num_tqubits):
-            self.add_tqubit("data")
-
-    def add_tqubit(self, sub_register: str) -> None:
-        if sub_register not in self.tqubits:
-            self.tqubits[sub_register] = {}
-        self.tqubits[sub_register][self.n] = self.tqubit_type(
-            params=self.params, name=self.name + "_" + str(self.n), circ=self.circ
-        )
-        self.n += 1
-
-    def add_tqubits(self, sub_register: str, num_tqubits: int) -> None:
-        for _ in range(num_tqubits):
-            self.add_tqubit(sub_register)
-
+        self.add_tqubits("data", ctype, params)
+    
+    def add_tqubits(
+        self, 
+        sub_register: str, 
+        ctype: List[str] = [XXZZ], 
```

Comment:
Would it be possible to just use the type rather than string constants and/or str(XXZZQubit) if we must use the string? It would cut down on having additional constants to maintain if we expand to other qubits. What do you think @Phionx?